modules/programs/neovim.nix

```diff
@@ -80,19 +61,29 @@ let
     optional = false;
   };
 
+  luaPackages = cfg.finalPackage.unwrapped.lua.pkgs;
+  resolvedExtraLuaPackages = cfg.extraLuaPackages luaPackages;
+
   extraMakeWrapperArgs = lib.optionalString (cfg.extraPackages != [ ])
     ''--suffix PATH : "${lib.makeBinPath cfg.extraPackages}"'';
-  extraMakeWrapperLuaCArgs = lib.optionalString (cfg.extraLuaPackages != [ ]) ''
-    --suffix LUA_CPATH ";" "${
-      lib.concatMapStringsSep ";" pkgs.lua51Packages.getLuaCPath
-      cfg.extraLuaPackages
-    }"'';
-  extraMakeWrapperLuaArgs = lib.optionalString (cfg.extraLuaPackages != [ ]) ''
-    --suffix LUA_PATH ";" "${
-      lib.concatMapStringsSep ";" pkgs.lua51Packages.getLuaPath
-      cfg.extraLuaPackages
-    }"'';
-
+  extraMakeWrapperLuaCArgs =
+    lib.optionalString (resolvedExtraLuaPackages != [ ]) ''
+      --suffix LUA_CPATH ";" "${
+        lib.concatMapStringsSep ";" luaPackages.getLuaCPath
+        resolvedExtraLuaPackages
+      }"'';
+  extraMakeWrapperLuaArgs = lib.optionalString (resolvedExtraLuaPackages != [ ])
+    ''
+      --suffix LUA_PATH ";" "${
+        lib.concatMapStringsSep ";" luaPackages.getLuaPath
+        resolvedExtraLuaPackages
+      }"'';
+
+  convertListToFunctionWithWarning = optionName:
+    flip warn const ''
+      Assigning a plain list to ${optionName} is deprecated.
+      Please assign a function taking a package set as argument instead.
```

Comment:
Can you improve the warning with an example please ? 
aka
`extraLuaPackages = [ pkgs.lua51Packages.basexx ]; should become extraLuaPackages = ls: [ ls.basexx ];`

Reply:
Ok, I added extra details in the description of the option, referring also to the example already provided there, and I referred to the option's documentation in the warning message, so that we do not duplicate the same information multiple times.
I think the option's description is the right place for this, so that people will see this info as soon as they look at the option, instead of discovering it only after having used the wrong type.

What do you think?

Reply:
I think it's easier if the example jumps at the user's face in the warning. Most of them dont want to look into the manual and just want to go on with their lives. Also your changes make the description quite verbose, no need to explain something that's going to be removed in the future.
e.g. I would remove
````
          For backwards compatibility, this option also accepts a plain list of
          packages, but this functionality is deprecated and might be removed in
          a future version, because doing so might cause problems when the
          specified packages are not compatible with the Python 3 version used
          by Neovim.
          ```

Reply:
Done.